### ISSUE TITLE
`invalid-method-override` false positive when overriding method that uses `Callable` with a paramspec

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ty#1820]
### Summary

```py
from typing import override, Callable
class Foo:
    def foo[**P](self, fn: Callable[P, None]): ...


class Bar(Foo):
    @override
    def foo[**P](self, fn: Callable[P, None]): ... # error: invalid-method-override
```

https://play.ty.dev/c17aa583-3631-41b3-be6e-79a86d91ac3b

### Version

0.0.1-alpha.32

**Comment by carljm** [comment]
Thanks for the report!

Verified that the issue doesn't repro with e.g.

```py
from typing import Callable

class Foo:
    def foo[T](self, fn: Callable[[T], None]): ...

class Bar(Foo):
    def foo[T](self, fn: Callable[[T], None]): ...
```

So it seems like an issue with callable-type assignability that's specific to ParamSpec. @dhruvmanila could you take a look?

**Comment by dalcinl** [comment]
I'm getting the same failure with the following reproducer (conde is in `*.pyi` stub). Most likely the same issue.

```python
from concurrent.futures import Executor, Future
from typing import Callable, ParamSpec, TypeVar
from typing import override

_P = ParamSpec("_P")
_T = TypeVar("_T")

class MyExecutor(Executor):
    def submit(
        self,
        fn: Callable[_P, _T],
        /,
        *args: _P.args,
        **kwargs: _P.kwargs,
    ) -> Future[_T]: ...
```

**Comment by dhruvmanila** [comment]
I'm following the trail of the assignability check:

| `self` | `target` |
|--------|--------|
| `Bar.foo` | `Foo.foo` |
| `Callable[P@foo, None]` | `Callable[P@foo, None]` |
| `P@foo.args` | `P@foo.args` |
| `tuple[object, ...]` | `P@foo.args` |

The reason for the final row is that `P.args` has an upper bound of `tuple[object, ...]` which is done by [this branch](https://github.com/astral-sh/ruff/blob/24ed28e31434106e3d0bcc8d1a1ede50b645c5da/crates/ty_python_semantic/src/types.rs#L2174-L2205). Pyright also fails at the [final assignability check](https://pyright-play.net/?strict=true&code=CYUwZgBAdg9g5gbQFRIAoF0AUAPAXBAFwFcAHAGxARgCMArEAYwIBoIA6D9VpAQwCc4AZ3yo2-IdyQBrAO7jhEUbPkBKXACgIWiPIgBeCNiA) but it might be special casing the `ParamSpec` somehow.

**Comment by AlexWaygood** [comment]
If we can't figure out a principled fix in time for the beta, I'd be fine with a quick short-term hack that skips Liskov checks for methods that have ParamSpecs in their generic context. Our Liskov checks are far from comprehensive currently anyway

**Comment by carljm** [comment]
@dhruvmanila It looks to me like we need special-cased handling for ParamSpec assignability to another ParamSpec, rather than falling back to the generalized upper bound like that?
